ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh

```diff
@@ -22,38 +22,80 @@ function config_samples_operator()
 }
 
-function mirror_tag_images () {
-    echo "quay.io/openshift-qe-optional-operators/ocp4-index:latest=MIRROR_REGISTRY_PLACEHOLDER/openshift-qe-optional-operators/ocp4-index:latest
-registry.redhat.io/ubi8/ruby-30:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi8/ruby-30:latest
-registry.redhat.io/ubi8/ruby-27:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi8/ruby-27:latest
-registry.redhat.io/ubi7/ruby-27:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi7/ruby-27:latest
-registry.redhat.io/rhscl/ruby-25-rhel7:latest=MIRROR_REGISTRY_PLACEHOLDER/rhscl/ruby-25-rhel7:latest
-registry.redhat.io/rhscl/mysql-80-rhel7:latest=MIRROR_REGISTRY_PLACEHOLDER/rhscl/mysql-80-rhel7:latest
-registry.redhat.io/rhel8/mysql-80:latest=MIRROR_REGISTRY_PLACEHOLDER/rhel8/mysql-80:latest
-registry.redhat.io/rhel8/httpd-24:latest=MIRROR_REGISTRY_PLACEHOLDER/rhel8/httpd-24:latest
-quay.io/openshifttest/osus-graph-data-container:latest=MIRROR_REGISTRY_PLACEHOLDER/openshifttest/osus-graph-data-container:latest
-" > /tmp/tag_images_list
-
+function config_mirror_auth () {
     run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
     if [[ $ret -eq 0 ]]; then
         echo "Get the cluster global pull secret successfully."
     else
         echo "!!! Fail to get the cluster global pull secret"
         return 1
     fi
+    # quay.io
+    optional_auth_user=$(cat "/var/run/vault/mirror-registry/registry_quay.json" | jq -r '.user')
+    optional_auth_password=$(cat "/var/run/vault/mirror-registry/registry_quay.json" | jq -r '.password')
+    quay_auth=`echo -n "${optional_auth_user}:${optional_auth_password}" | base64 -w 0`
+    # # brew.registry.redhat.io
+    # reg_brew_user=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.user')
+    # reg_brew_password=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.password')
+    # brew_registry_auth=`echo -n "${reg_brew_user}:${reg_brew_password}" | base64 -w 0`
+    # vmc.mirror-registry.qe.devcluster.openshift.com:5000
+    registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+    # registry.redhat.io
+    reg_user=$(cat "/var/run/vault/mirror-registry/registry.json" | jq -r '.user')
+    reg_password=$(cat "/var/run/vault/mirror-registry/registry.json" | jq -r '.password')
+    registry_auth=`echo -n "${reg_user}:${reg_password}" | base64 -w 0`
 
-    sed -i "s/MIRROR_REGISTRY_PLACEHOLDER/${REGISTRY_HOST}:5000/g" "/tmp/tag_images_list"
-    run_command "cat /tmp/tag_images_list"
-    run_command "oc image mirror -f \"/tmp/tag_images_list\"  --insecure=true -a \"/tmp/.dockerconfigjson\" --skip-missing=true --skip-verification=true --keep-manifest-list=true --filter-by-os='.*'"; ret=$?
+    jq --argjson a "{\"registry.redhat.io\": {\"auth\": \"${registry_auth}\"}, \"quay.io\": {\"auth\": \"$quay_auth\"}, \"${MIRROR_PROXY_REGISTRY}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "/tmp/.dockerconfigjson" > /tmp/new-dockerconfigjson
```

Comment:
I just go though the installer mirror steps, seem like we do not have to get pull secret for quay.io, redhat.io from vault, all the pull secrets are already  saved in `${CLUSTER_PROFILE_DIR}/pull-secret`, hence we can directly use `${CLUSTER_PROFILE_DIR}/pull-secret` for mirror command, once that, we can drop `config_mirror_auth`.

Reply:
Yeah, but remaining this `config_mirror_auth` will make this step doesn't depend on any other steps.

Reply:
Anyway, updated it.